arch/sim/src/sim/up_i2cbuslinux.c

```
@@ -0,0 +1,310 @@
+
+#define ERROR(fmt, ...) \
+        syslog(LOG_ERR, "up_i2cbuslinux: " fmt "\n", ##__VA_ARGS__)
+#define INFO(fmt, ...) \
+        syslog(LOG_ERR, "up_i2cbuslinux: " fmt "\n", ##__VA_ARGS__)
+#define DEBUG(fmt, ...)
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+struct linux_i2cbus_master_s
+{
+  const struct i2c_ops_s *ops; /* I2C vtable */
+  int file;
+};
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int linux_i2cbus_transfer(struct i2c_master_s *dev,
+                                 struct i2c_msg_s *msgs, int count);
+#ifdef CONFIG_I2C_RESET
+static int linux_i2cbus_reset(struct i2c_master_s *dev);
+#endif
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static struct i2c_ops_s i2c_linux_ops =
+{
+  .transfer = linux_i2cbus_transfer,
+#ifdef CONFIG_I2C_RESET
+  .reset = linux_i2cbus_reset,
+#endif
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: linux_i2cbus_reset
+ *
+ * Description:
+ *   Provide i2c reset
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_I2C_RESET
+static int linux_i2cbus_reset(struct i2c_master_s *dev)
+{
+  return -1; /* Not implemented */
```

-NOSYS

---

Do we guarantee that the error numbers map?  I thought we just had to throw a generic -1 across the boundary?

---

Even the error number isn't exactly match between Linux and NuttX, it is always better to return the difference error code than -1. Eating the real error code just make the debugging more tough.